async_producer.go

```diff
@@ -555,6 +555,21 @@ func (tp *topicProducer) partitionMessage(msg *ProducerMessage) error {
 		return err
 	}
 
+	// Select only partitions with leaders in this rack if configured so, falling back if none are available.
+	if tp.parent.conf.Producer.PartitionerRackAware {
+		var availablePartitionsInRack []int32
+		var clientRack = tp.parent.client.Config().RackID
+		for _, p := range partitions {
+			leaderBroker, err := tp.parent.client.Leader(msg.Topic, p)
+			if err == nil && leaderBroker.Rack() == clientRack {
+				availablePartitionsInRack = append(availablePartitionsInRack, p)
+			}
```

Comment:
🤔 Go style tends to prefer unintending the non-error flow, but I don’t think this is a case where it’s altogether worth it, but here’s the alternative:
```suggestion
			if leader, err := tp.parent.client.Leader(msg.Topic, p); err != nil || leader.Rack() != clientRack {
				continue
			}
			availablePartitionsInRack = append(availablePartitionsInRack, p)
```

N.B.: Even without using the `if … { continue }`, I would still recommend using just `leader` as that’s what the function is called. In the alternative, `leadBroker` is a more natural rendition of “broker which is also the leader”.

Reply:
Thanks for the feedback, Cassondra! I'll keep the indent with your comment in mind.